File: 1342/c++/main.cpp
#include <iostream>

using namespace std;

int numberOfSteps(int num)
{
    int res = 0;

    while (num > 0)
    {
        if ((num & 1) == 1)
        {
            num = num ^ 1;
        }
        else
        {
            num = num >> 1;
        }
        res++;
    }

    return res;
}

int main()
{
    cout << numberOfSteps(14) << endl;
}

File: 1342/main.go
package main

import (
	"fmt"
)

func numberOfSteps(num int) int {
	res := 0

	for num > 0 {
		if num&1 == 1 {
			num = num ^ 1
		} else {
			num = num >> 1
		}
		res++
	}

	return res
}

func main() {
	fmt.Println(numberOfSteps(14))
}
